RecoEgamma/EgammaIsolationAlgos/src/EgammaHadTower.cc

```diff
@@ -111,33 +123,50 @@ double EgammaHadTower::getDepth2HcalESum(const std::vector<CaloTowerDetId> & tow
 }
 
 bool EgammaHadTower::hasActiveHcal( const std::vector<CaloTowerDetId> & towers ) const {
-  bool debug = false; // change this to true to get debug output
+
   bool active = false;
   int statusMask = ((1<<HcalChannelStatus::HcalCellOff) | (1<<HcalChannelStatus::HcalCellMask) | (1<<HcalChannelStatus::HcalCellDead));
-  if (debug) std::cout << "DEBUG: hasActiveHcal called with " << towers.size() << " detids. First tower detid ieta " << towers.front().ieta() << " iphi " << towers.front().iphi() << std::endl;
+#ifdef EDM_ML_DEBUG
+  std::cout << "DEBUG: hasActiveHcal called with " << towers.size() 
+	    << " detids. First tower detid ieta " << towers.front().ieta() 
+	    << " iphi " << towers.front().iphi() << std::endl;
+#endif
   for (auto towerid : towers) {
       unsigned int ngood = 0, nbad = 0;
       for (DetId id : towerMap_->constituentsOf(towerid)) {
           if (id.det() != DetId::Hcal) {
               continue;
           }
           HcalDetId hid(id);
-          if (debug) std::cout << "      hcal constituent on subdet " << hid.subdet() << ", ieta " << hid.ieta() << " iphi " << hid.iphi() << ", depth " << hid.depth() << std::endl;
           if (hid.subdet() != HcalBarrel && hid.subdet() != HcalEndcap) continue;
-          const auto * values = hcalQuality_->getValues(id,/*throwOnFail=*/false);
-          if (!values) {
-              if (debug) std::cout << "          MISSING IN CONDITIONS" << std::endl;
-              continue;
-          }
-          int status = values->getValue();
+#ifdef EDM_ML_DEBUG
+          std::cout << "EgammaHadTower DetId " << std::hex << id.rawId() 
+		    << "  hid.rawId  " << hid.rawId() << std::dec   
+		    << "   sub " << hid.subdet() << "   ieta " << hid.ieta() 
+		    << "   iphi " << hid.iphi() 
+		    << "   depth " << hid.depth() << std::endl;
+#endif
+	  // Sunanda's fix for 2017 Plan1   
+          // and removed protection 
+          int status = hcalQuality_->getValues((DetId)(hcalTopology_->idFront(HcalDetId(id))),/*throwOnFail=*/false)->getValue();
```

Comment:
shouldn't `throwOnFail` now be set to `true` instead?
otherwise, we'd get a crash instead of an exception.